include/emp/prefab/ToggleButtonGroup.hpp

```diff
@@ -0,0 +1,243 @@
+        handle_toggle(true);
+      });
+      // OnClick used due to a strange bug(?) in which the input radios only
+      // fire their onchange function once
+      // Probably due to Bootstrap/jQuery weirdness and this is a last minute
+      // work around ¯\_(ツ)_/¯
+
+      web::Input activate_radio(
+        [](std::string){ ; },
+        "radio", "", emp::to_string(GetID(), "_activate_radio"),
+        false, false
+      );
+      activate_label << activate_radio;
+      activate_label << std::forward<L1_TYPE>(activate_indicator);
+
+      deactivate_label.AddAttr(
+        "class", "active",
+        "class", "btn",
+        "class", emp::to_string("btn-outline-", deactivate_style)
+      );
+      deactivate_label.OnClick([tog=*this, &handle_toggle=on_toggle]() mutable {
+        tog.SetActive(false);
+        handle_toggle(false);
+      });
+
+      web::Input deactivate_radio(
+        [](std::string){ ; },
+        "radio", "", emp::to_string(GetID(), "_deactivate_radio"),
+        false, true
+      );
+      deactivate_label << deactivate_radio;
+      deactivate_label << std::forward<L2_TYPE>(deactivate_indicator);
+    }
+
+    public:
+    /**
+     * @param activate_indicator a string, FontAwesomeIcon or other component
+     * indicating that the first button actives this toggle
+     * @param deactivate_indicator a string, FontAwesomeIcon or other component
+     * indicating that the second button deactivates this toggle
+     * @param activate_style a bootstrap style (primary, secondary, etc) for
+     * the first button
+     * @param deactivate_style a bootstrap style (primary, secondary, etc) for
+     * the second button
+     * @param cassette_style whether the toggle should display in cassette style
+     * (both buttons visible) or do a swap on toggle (one button visible)
+     * @param grayout in cassette mode, whether buttons should be
+     * grayed out to further emphasize the current state
+     * @param in_id HTML ID of ToggleButtonGroup div
+     */
+    template<typename L1_TYPE, typename L2_TYPE>
+    ToggleButtonGroup(
+      L1_TYPE && activate_indicator,
+      L2_TYPE && deactivate_indicator,
+      const std::string & activate_style="success",
+      const std::string & deactivate_style="warning",
+      const bool & cassette_style=true,
+      const bool & grayout=false,
+      const std::string & in_id=""
+    ) : ToggleButtonGroup(
+      std::forward<L1_TYPE>(activate_indicator),
+      std::forward<L2_TYPE>(deactivate_indicator),
+      activate_style, deactivate_style,
+      cassette_style, grayout,
+      new internal::ToggleButtonGroupInfo(in_id)
+    ) { ; }
+
+    /**
+     * Determines whether the toggle is in the active state
+     * @return boolean
+     */
+    bool IsActive() const {
+      return Info()->IsActive();
+    }
+
+    /**
+     * Set the active state of the toggle
+     * @param is_active a boolean
+     */
+    void SetActive(const bool & val) {
```

Comment:
ibid, SetActive and SetInactive

Reply:
I'm just going to remove this method since I'm not sure it's currently possible to update styles correctly to match state.